src/mscorlib/src/System/Text/ASCIIEncoding.cs

```diff
@@ -25,9 +25,11 @@ namespace System.Text
 [System.Runtime.InteropServices.ComVisible(true)]
     public class ASCIIEncoding : Encoding
     {
+        // Allow for devirtualization (see https://github.com/dotnet/coreclr/pull/9230)
+        internal sealed class ASCIIEncodingSealed : ASCIIEncoding { }
```

Comment:
Ah, yes. The type will both need to be annotated add Serializable and have a serialization ctor added.